internal/service/ipam/api_address_block_resource.go

```diff
@@ -0,0 +1,168 @@
+package ipam
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+
+	bloxoneclient "github.com/infobloxopen/bloxone-go-client/client"
+)
+
+// Ensure provider defined types fully satisfy framework interfaces.
+var _ resource.Resource = &AddressBlockResource{}
+var _ resource.ResourceWithImportState = &AddressBlockResource{}
+
+func NewAddressBlockResource() resource.Resource {
+	return &AddressBlockResource{}
+}
+
+// AddressBlockResource defines the resource implementation.
+type AddressBlockResource struct {
+	client *bloxoneclient.APIClient
+}
+
+func (r *AddressBlockResource) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_" + "ipam_address_block"
+}
+
+func (r *AddressBlockResource) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		MarkdownDescription: "",
```

Comment:
Not for this PR, but we will have to update this for all resources later.

Reply:
Is this description ok: `Handles the creation, updation, and deletion of Address Block object`?

Reply:
No you have to give the more verbose description that applies to the address block object itself. For e.g. the below description is copied from the swagger doc 

`The AddressBlock object allows a uniform representation of the address space segmentation, supporting functions such as administrative grouping, routing aggregation, delegation etc.`